stan/math/rev/mat/fun/sqrt_spd.hpp

```diff
@@ -35,19 +35,23 @@ class sqrt_spd_op {
   //  derivative-or-differential-of-symmetric-square-root-of-a-matrix
 
   template <std::size_t size>
-  std::tuple<Eigen::VectorXd>
+  std::tuple<Eigen::MatrixXd>
   multiply_adjoint_jacobian(const std::array<bool, size> &needs_adj,
-                            const Eigen::VectorXd &adj) const {
+                            const Eigen::MatrixXd &adj) const {
     using Eigen::kroneckerProduct;
     using Eigen::MatrixXd;
+    Eigen::MatrixXd output(K_, K_);
+    Eigen::Map<const Eigen::VectorXd> map_input(adj.data(), adj.size());
```

Comment:
And if `adj` is a `MatrixXd`, how can it be mapped to a `VectorXd`?

Reply:
That mapping is doing this (https://en.wikipedia.org/wiki/Vectorization_(mathematics)). I was just pattern-matching the eqs in the Stackoverflow link. I'm not really familiar with that. I was just double checking myself and Abhishek makes a note at the end of his post that the Jacobian he wrote is in terms of the vectorization, which I assume is the Wikipedia definition.

Reply:
Oh yeah fair warning, I was just kinda guessing on the math stuff to make things compile. Maybe add in a diagonal matrix test (with things that aren't 1 on the diagonal) so we know the answer and gradients and all just to make sure things are in the right direction.